tools/crmadmin.c

```diff
@@ -119,14 +118,56 @@ static GOptionEntry additional_options[] = {
       "Display nodes as shell commands of the form 'export uname=uuid'"
       "\n                          (valid with -N/--nodes)",
     },
-    { "ipc-name", 'i', G_OPTION_FLAG_HIDDEN, G_OPTION_ARG_STRING, &ipc_name,
+    { "ipc-name", 'i', 0, G_OPTION_ARG_STRING, &ipc_name,
       "Name to use for ipc instead of 'crmadmin' (with -P/--pacemakerd).",
       NULL
     },
 
     { NULL }
 };
 
+gboolean
+command_cb(const gchar *option_name, const gchar *optarg, gpointer data, GError **error)
+{
+    if (!strcmp(option_name, "--status") || !strcmp(option_name, "-S")) {
+        command = cmd_health;
+        crm_trace("Option %c => %s", 'S', optarg);
+    }
+
+    if (!strcmp(option_name, "--pacemakerd") || !strcmp(option_name, "-P")) {
+        command = cmd_pacemakerd_health;
+        need_pacemakerd_api = true;
+        need_controld_api = false;
+    }
```

Comment:
For efficiency these should all be "else if", otherwise it will do all the strcmp()s even if one matches early. Efficiency isn't really a concern for argument processing, but it helps to get in the habit